test/e2e/annotations/authtls.go

```diff
@@ -335,7 +334,7 @@ var _ = framework.DescribeAnnotation("auth-tls-*", func() {
 		annotations := map[string]string{
 			"nginx.ingress.kubernetes.io/auth-tls-secret":        nameSpace + "/" + host,
 			"nginx.ingress.kubernetes.io/auth-tls-verify-client": "on",
-			"nginx.ingress.kubernetes.io/auth-tls-match-cn":      "CN=(itwillmatch|withthenextoption|authtls)",
```

Comment:
Is removing the CN= requirement a breaking change for this annotation? Or it will still works? I'm worried about removing this and breaking the behavior.

Reply:
hmm I think this may be tricky, I was asked by my team mate here: https://github.com/kubernetes/ingress-nginx/pull/8434#discussion_r862952023, after the PR was already merged that this "CN=" part is not the intended functionality we were aiming for.

Let me know if you have any ideas, because the way I have it built now its basically grabbing the CN without needing to prefix it with "CN="

Reply:
Just retested with the "CN=" part added again and everything passes!

No changes will be required for this and will not cause breaks thanks to the !~ (does not regex match), which gives the check that little extra flexibility we want in auth-tls-match-cn.

I still think that for this test though it makes sense to drop the "CN=" part as with the new way that we are capturing the CN directly from ssl_clien_s_dn, it just adds some confusion.